lib/account.rb

```diff
@@ -0,0 +1,73 @@
+require 'csv'
+
+module Bank
+  class Account
+    attr_reader :id, :balance, :open_date, :count
+    @@all_accounts = []
+
+    def initialize(id, balance, open_date = '2012 - 12-21 12:21:12 -0800')
+      @id = id
+      @balance = balance
+      raise ArgumentError, "You cannot have a negative balance" unless balance >= 0
+      @open_date = open_date
+      # @@all_accounts << self
+    end
+
+    def self.all
+      @@all_accounts ||= read_csv
```

Comment:
We talked about this in class, but I really like this solution - it's a clever way to avoid reading the file multiple times if `Account.all` is called more than once.